FakeReader: link instrument to all images

sbesson opened the pull request:
Noticed during some internal testing, the mock instrument created in `FakeReader` either when using `plates=N` or when setting `withInstrument=true` is completely unreferenced in the OME metadata. This commit updates the metadata to create:
- Image.InstrumentRef and Image.ObjectiveSettings for each image
- Channel.DetectorSettings, Channel.FilterRefSet, Channel.LightSourceSettings for each channel of each image
- LightPath.DichroicRef, LightPath.EmissionFilterRef, LightPath.ExcitationFilterRef for the LightPath element of each channel of each image

The unit tests are updated to test existence of the various elements for all representations (single image, multi-image and SPW, single and multi-channel)

Functionally, this PR can be tested by running

```
showinf -nopix -omexml-only "test&series=2&withInstrument=true.fake"
showinf -nopix -omexml-only "test&series=2&sizeC=5&withInstrument=true.fake"
showinf -nopix -omexml-only "test&plates=1&plateRows=2&plateCols=2.fake"
showinf -nopix -omexml-only "test&plates=1&plateRows=2&plateCols=2&sizeC=3.fake"
```

Without this PR, the `Image` elements should have no reference to the `Instrument`, `Detector`... With this PR, all the elements under the `Instrument:0` should be linked in the relevant location.

sbesson committed:
FakeReader: link instrument to all images

Update unit tests to test the link existence for all representations (single image, multi-image and SPW)

melissalinkert commented:
Would it make sense to also link the objective to the image(s) and/or link any of the light sources to channels?

sbesson committed:
Link all Instrument elements appropriately

Add references for detector, objective, dichroic, filter, filter set
and light sources
Expand unit tests to ensure that all objects are populated

sbesson commented:
> Would it make sense to also link the objective to the image(s) and/or link any of the light sources to channels?

👍 Last commit should now link the Instrument and all subelements in the relevant place of the OME metadata graph. The unit tests and the PR description have also been updated accordingly

melissalinkert reviewed (approved):
https://bf-testing-results.s3.amazonaws.com/2025/2025-08-16/build-status.log
